src/hosted_db/sqlite.rs

```diff
@@ -36,3 +47,35 @@ pub fn run_sqlite_query(path: &PathBuf, query: &str) -> Result<QueryResult, AybE
         rows: results,
     })
 }
+
+pub async fn run_sqlite_query(
+    path: &PathBuf,
+    query: &str,
+    isolation: &Option<AybConfigIsolation>,
+) -> Result<QueryResult, AybError> {
+    match isolation {
+        Some(isolation) => {
+            let result = run_in_sandbox(Path::new(&isolation.nsjail_path), path, query).await?;
+
+            if result.stderr.len() > 0 {
+                let error: AybError = serde_json::from_str(&result.stderr)?;
+                Err(error)
+            } else if result.status != 0 {
+                Err(AybError::Other {
+                    message: format!(
+                        "Error status from sandboxed query runner: {}",
+                        result.status.to_string()
+                    ),
+                })
+            } else if result.stdout.len() > 0 {
+                let query_result: QueryResult = serde_json::from_str(&result.stdout)?;
+                Ok(query_result)
+            } else {
+                Err(AybError::Other {
+                    message: "No results from sandboxed query runner".to_string(),
+                })
+            }
+        }
+        None => Ok(query_sqlite(path, query)?.into()),
```

Comment:
You could remove a level of nesting by doing the following at the top
```rust
if isolation.is_none() {
   return Ok(query_sqlite(path, query)?.into());
}
```

Reply:
I'm conflicted on this one. We'd get one level of nesting in a relatively short function, but in exchange
* I would stop using a language feature like `match`, which I feel is being used appropriately
* I'd have to introduce an `unwrap`, since the compiler won't be able to reason about the knowledge we'd gain with the `if`

Given we're using match in the way it was intended and forcing ourselves to enumerate all of the variants, I wonder if we'd gain much in exchange for reducing indentation in a short function.

Reply:
This is just a style preference, the `Option` enum won't add more variants and thus `match` doesn't introduce any advantages over other ways of error checking/handling.

I feel that you are _really really_ skeptical about using `unwrap`/`expect`. Let's not forget that in our case when something panics inside a request to the server, the ayb server keeps running. In our case, using `unwrap`/`expect` in a way that can _actually_ panic isn't as bad as using it in a library or a TUI (not saying that that's the case with this suggestion, this is just a general comment).

I like early returns because they remove mental overhead from the developer and clearly separate the "invalid cases"/error handling from the actual logic of the code.

Reply:
New idea! What about using `if let Some(isolation) = isolation` and returning the `None` case at the end of the function? Seems like a reasonable middleground :)

```rust
pub async fn potentially_isolated_sqlite_query(
    path: &PathBuf,
    query: &str,
    isolation: &Option<AybConfigIsolation>,
) -> Result<QueryResult, AybError> {
    if let Some(isolation) = isolation {
        let result = run_in_sandbox(Path::new(&isolation.nsjail_path), path, query).await?;
        if !result.stderr.is_empty() {
            let error: AybError = serde_json::from_str(&result.stderr)?;
            return Err(error);
        }
        
        if result.status != 0 {
            return Err(AybError::Other {
                message: format!(
                    "Error status from sandboxed query runner: {}",
                    result.status
                ),
            });
        }
        
        if !result.stdout.is_empty() {
            let query_result: QueryResult = serde_json::from_str(&result.stdout)?;
            return Ok(query_result);
        }
        
        return Err(AybError::Other {
            message: "No results from sandboxed query runner".to_string(),
        });
    }
    
    Ok(query_sqlite(path, query)?)
}
```

Reply:
Clean! I like it! I appreciated both of your bullet points on the broader point. I'll implement this and we should be good to go! Woohoo! Thank you!!!